testsuite/MDAnalysisTests/analysis/test_hbonds.py

```diff
@@ -183,65 +184,74 @@ def _run(self, **kwargs):
             h.run(verbose=False)
         return h
 
-    def test_check_static_selections(self):
+    def run_HBA(self, s1, s2, s1type):
+        """test that HydrogenBondAnalysis() raises SelectionError for missing donors/acceptors"""
+        # no donors/acceptors; only raises error if no updates
+        return self._run(
+            selection1=s1,
+            selection2=s2,
+            update_selection1=False,
+            update_selection2=False,
+            selection1_type=s1type,
+        )
+
+    @pytest.mark.parametrize('s1, s2, s1type', itertools.product(
+        ("protein", "resname ALA and not backbone"),
+        ("protein", "resname ALA and not backbone"),
+        ("donor", "acceptor", "both")
+    ))
+    def test_check_static_selections_pt(self, s1, s2, s1type):
         self._setUp()
-        try:
-            def run_HBA(s1, s2, s1type):
-                """test that HydrogenBondAnalysis() raises SelectionError for missing donors/acceptors"""
-                # no donors/acceptors; only raises error if no updates
-                return self._run(selection1=s1, selection2=s2,
-                                 update_selection1=False, update_selection2=False,
-                                 selection1_type=s1type,
-                )
-            protein = "protein"
-            nothing = "resname ALA and not backbone"
-            for s1, s2, s1type in itertools.product((protein, nothing),
-                                                    (protein, nothing),
-                                                    ("donor", "acceptor", "both")):
-                if s1 == s2 == protein:
-                    def runOK():
-                        """test that HydrogenBondAnalysis() works for protein/protein"""
-                        try:
-                            h = run_HBA(s1, s2, s1type)
-                        except:
-                            raise AssertionError("HydrogenBondAnalysis protein/protein failed")
-                        else:
-                            return True
-                    yield runOK
+        protein = "protein"
+
+        if s1 == s2 == protein:
+            def runOK():
+                """test that HydrogenBondAnalysis() works for protein/protein"""
+                try:
+                    h = self.run_HBA(s1, s2, s1type)
+                except:
+                    raise pytest.fail("HydrogenBondAnalysis protein/protein failed")
                 else:
-                    yield assert_raises, SelectionError, run_HBA, s1, s2, s1type
-        finally:
-            self._tearDown()
-
-    def test_run_empty_selections(self):
-        self._setUp()
+                    return True
+
+            runOK()
+        else:
+            with pytest.raises(SelectionError):
+                self.run_HBA(s1, s2, s1type)
+
+        self._tearDown()
+
+    def run_HBA2(self, s1, s2, s1type):
```

Comment:
I do not like having functions named `run_HBA` and `run_HBA2`. It was OK that they had the same name when they were nested because the parent function was giving the context. Now the names do not tell how the two functions differ from each other. `run_HBA_no_update` and `run_HBA_update` would be better names, other names may work as well.